apps/explorer/src/ui/ExpandableList.tsx

```diff
@@ -26,30 +80,28 @@ export function ExpandableList({
     const handleShowAllClick = () =>
         setShowAll((prevShowAll: boolean) => !prevShowAll);
 
+    let showAllText = '';
+    if (showAll) {
+        showAllText = 'Show Less';
+    } else {
+        showAllText = itemsLabel
+            ? `Show All ${items.length} ${itemsLabel}`
+            : 'Show All';
+    }
+
     return (
-        <>
-            {itemsDisplayed.map((item, index) => (
-                <div key={index}>{item}</div>
-            ))}
-            {items.length > defaultItemsToShow && (
-                <div className="mt-4 flex cursor-pointer items-center gap-1 text-steel hover:text-steel-dark">
-                    <Link
-                        variant="text"
-                        onClick={handleShowAllClick}
-                        after={
-                            <ChevronDown12
-                                height={12}
-                                width={12}
-                                className={showAll ? 'rotate-180' : ''}
-                            />
-                        }
-                    >
-                        <Text variant="bodySmall/medium">
-                            {showAll ? 'Show Less' : 'Show All'}
-                        </Text>
-                    </Link>
-                </div>
-            )}
-        </>
+        <ExpandableListContext.Provider
+            value={{
+                handleShowAllClick,
+                showAll,
+                showAllText,
+                itemsDisplayed: itemsDisplayed.map((item, index) => (
+                    <div key={index}>{item}</div>
```

Comment:
I know this is copied over but why do we do this?

Reply:
Also you should just put all of the items in context then let the `ExpandableListItems` handle showing a subset of it. The stuff you should put in context is basically the limit, the list of items, and if you're showing all or not. The child components can handle the rest.